music_assistant/providers/radioparadise/__init__.py

```diff
@@ -0,0 +1,369 @@
+        self._current_stream_details: StreamDetails | None = None
+        self._monitor_task: asyncio.Task[None] | None = None
+
+    @property
+    def supported_features(self) -> set[ProviderFeature]:
+        """Return the features supported by this Provider."""
+        return {
+            ProviderFeature.BROWSE,
+            ProviderFeature.LIBRARY_RADIOS,
+            ProviderFeature.TRACK_METADATA,
+        }
+
+    @property
+    def is_streaming_provider(self) -> bool:
+        """Return True if the provider is a streaming provider."""
+        return True
+
+    async def loaded_in_mass(self) -> None:
+        """Call after the provider has been loaded."""
+        if self.mass and hasattr(self.mass, "music") and hasattr(self.mass.music, "sync_library"):
+            self.mass.create_task(self.mass.music.sync_library(self.instance_id, MediaType.RADIO))
+
+    async def get_library_radios(self) -> AsyncGenerator[Radio, None]:
+        """Retrieve library/subscribed radio stations from the provider."""
+        for channel_id in self._channels_cache:
+            yield await self._parse_radio(channel_id)
+
+    async def get_radio(self, prov_radio_id: str) -> Radio:
+        """Get full radio details by id."""
+        if prov_radio_id not in self._channels_cache:
+            raise MediaNotFoundError("Station not found")
+        return await self._parse_radio(prov_radio_id)
+
+    async def get_stream_details(self, item_id: str, media_type: MediaType) -> StreamDetails:
+        """Get streamdetails for a radio station."""
+        if media_type != MediaType.RADIO:
+            raise ValueError(f"Unsupported media type: {media_type}")
+        if item_id not in self._channels_cache:
+            raise ValueError(f"Unknown radio channel: {item_id}")
+
+        stream_url = self._build_stream_url(item_id)
+        if not stream_url:
+            raise ValueError(f"No stream URL found for channel {item_id}")
+
+        channel_info = self._channels_cache[item_id]
+        stream_format = next(
+            (k for k, v in channel_info["stream_urls"].items() if v == stream_url),
+            self._stream_format,
+        )
+        format_info = BITRATE_FORMATS.get(stream_format, BITRATE_FORMATS["flac"])
+
+        self._current_stream_details = StreamDetails(
+            item_id=item_id,
+            provider=self.lookup_key,
+            audio_format=AudioFormat(
+                content_type=cast("ContentType", format_info["content_type"]),
+                sample_rate=cast("int", format_info["sample_rate"]),
+                bit_depth=cast("int", format_info["bit_depth"]),
+                channels=2,
+            ),
+            media_type=MediaType.RADIO,
+            stream_type=StreamType.HTTP,
+            path=stream_url,
+            allow_seek=False,
+            can_seek=False,
+            duration=0,
+        )
+
+        self._monitor_task = self.mass.create_task(self._monitor_stream_metadata(item_id))
```

Comment:
I'm not a fan of this approach as it will only work for one stream at a time. If you can only do one stream at a time on purpose, then also guard this in the get streamdetails. Otherwise just spin up a new monitor task within the get_streamdetails